### PR TITLE
Support Fleet output Logstash ssl fields

### PR DESCRIPTION
**Pull request by daemitus**
re: #497 

Add in the 3 ssl fields to the logstash-type fleet output resource. 

I've run it on my dev stack a few times and I'm happy with most everything except the anonymous nested SSL struct that gets generated. Can that get yanked out?

### CODE REVIEW

File: internal/fleet/output_resource.go [hunk 1]
```diff
@@ -299,6 +340,23 @@ func resourceOutputUpdateLogstash(ctx context.Context, d *schema.ResourceData, m
 	if value, ok := d.Get("ca_sha256").(string); ok && value != "" {
 		reqData.CaSha256 = &value
 	}
+	if value, ok := d.GetOk("ssl"); ok {
+		ssl := value.([]interface{})[0].(map[string]interface{})
+		reqData.Ssl = &struct {
```
Comment (daemitus) [inline internal/fleet/output_resource.go:354]: Gross

File: internal/fleet/output_resource.go [hunk 2]
```diff
@@ -299,6 +340,23 @@ func resourceOutputUpdateLogstash(ctx context.Context, d *schema.ResourceData, m
...
+			Certificate            *string   `json:"certificate,omitempty"`
+			CertificateAuthorities *[]string `json:"certificate_authorities,omitempty"`
+			Key                    *string   `json:"key,omitempty"`
+		}{}
```
Comment (daemitus) [inline internal/fleet/output_resource.go:358]: Anonymous structs :(

**Comment by daemitus** [comment]
eh hold on. certificate_authorities isnt working.

File: internal/fleet/output_resource.go (hunk 2 above)
Comment (tobio) [reply internal/fleet/output_resource.go:358]: Yea, this is pretty naff. https://github.com/elastic/terraform-provider-elasticstack/issues/499

**Review by tobio — APPROVED** [review]
Thanks for taking the time to add this into the provider! Sorry the clients a bit nasty to work with right now :)

**Comment by tobio** [comment]
@daemitus are you able to take a look at the test failures here?

**Comment by daemitus** [comment]
Sure thing, I've got some time today.

File: internal/fleet/output_resource.go [hunk 3]
```diff
@@ -174,6 +203,18 @@ func resourceOutputCreateLogstash(ctx context.Context, d *schema.ResourceData, m
 	if value, ok := d.Get("ca_trusted_fingerprint").(string); ok && value != "" {
 		reqData.CaTrustedFingerprint = &value
 	}
+	if value, ok := d.GetOk("ssl"); ok {
+		ssl := value.([]interface{})[0].(map[string]interface{})
```
Comment (tobio) [inline internal/fleet/output_resource.go:207]: reqData.Ssl will be nil for all these assignments. You probably just want to set it to an empty struct here.

**Review by tobio — APPROVED** [review]
Nice one! Thanks for taking the time to get this one in :)

**Pull request merged**